salt/utils/dictupdate.py

```diff
@@ -50,7 +54,7 @@ def update(dest, upd, recursive_update=True, merge_lists=False):
             elif isinstance(dest_subkey, list) \
                      and isinstance(val, list):
                 if merge_lists:
-                    dest[key] = dest.get(key, []) + val
+                    dest[key] = sorted((set(dest_subkey + val)))
```

Comment:
So, if we're going to merge lists and preserve order, we _can_ gate that behind an argument. But the person using the code is going to need to meet us halfway on it. Both lists will need to be sorted prior to calling the merge function, for it to work.